docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1232 @@
+    @app_commands.guild_only()
+    async def serverinfo(interaction: discord.Interaction):
+        assert interaction.guild is not None
+        await interaction.response.send_message(interaction.guild.name)
+
+Default permissions
+++++++++++++++++++++
+
+This sets the default permissions a user needs in order to be able to see and invoke an app command.
+
+Configured by adding the :func:`.app_commands.default_permissions` decorator when defining an app command:
+
+.. code-block:: python
+
+    @client.tree.command()
+    @app_commands.default_permissions(manage_nicknames=True)
+    async def nickname(interaction: discord.Interaction, newname: str):
+        guild = interaction.guild
+        if not guild:
+            await interaction.response.send_message("i can't change my name here")
+        else:
+            await guild.me.edit(nick=newname)
+            await interaction.response.send_message(f"hello i am {newname} now")
+
+Commands with this check are still visible and invocable in the bot's direct messages,
+regardless of the permissions specified.
+
+To prevent this, :func:`~.app_commands.guild_only` can also be added.
+
+.. warning::
+
+    This can be overriden to a different set of permissions by server administrators through the "Integrations" tab on the official client,
+    meaning, an invoking user might not actually have the permissions specified in the decorator.
+
+Custom checks
+++++++++++++++
+
+waiting to be written
+
+cover:
+
+- how to make a check, what it should return, default behaviours
+- builtin common checks and exceptions
+
+Custom checks come in two forms:
+
+- A local check, which runs for a single command
+- A global check, which runs before all commands, and before any local checks
+
+Global check
+^^^^^^^^^^^^^
+
+To define a global check, override :meth:`.CommandTree.interaction_check` in a :class:`~.app_commands.CommandTree` subclass.
+This method is called before every command invoke.
+
+For example:
+
+.. code-block:: python
+
+    whitelist = {236802254298939392, 402159684724719617} # cool people only
+
+    class MyCommandTree(app_commands.CommandTree):
+        async def interaction_check(self, interaction: discord.Interaction) -> bool:
+            return interaction.user.id in whitelist
+
+.. note::
+
+    If your project uses :class:`.ext.commands.Bot` as the client instance,
+    the :class:`.CommandTree` class can be configured via
+    the ``tree_cls`` keyword argument in the bot constructor:
+
+    .. code-block:: python
+        :emphasize-lines: 6
+
+        from discord.ext import commands
+
+        bot = commands.Bot(
+            command_prefix="?",
+            intents=discord.Intents.default(),
+            tree_cls=MyCommandTree
+        )
+
+Error handling
+---------------
+
+So far, any exceptions raised within a command callback, any custom checks or in a transformer should just be
+printed out in the program's ``stderr`` or through any custom logging handlers.
```

Comment:
```suggestion
By default, exceptions are logged using the library logger if the command does not have any error handlers attached to it.
```
Errors are no longer printed to `stderr` by default since v2.0